React/CxxBridge/RCTCxxBridge.mm

```diff
@@ -372,25 +374,42 @@ - (void)start
     }));
   }
 
+  // TODO(OSS Candidate ISS#2710739) runtime initialization sequencing
+  [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(runtimeInitializationDidEnd:) name:(NSString*)RCTRuntimeInitializationEndNotificationName object:nil];
```

Comment:
Observing a global notification concerns me because we can have multiple runtimes being initialized concurrently. Also, shouldn't need to cast `RCTRuntimeInitializationEndNotificationName` to `NSString` here.

Reply:
I don't think in practice this would break the multi runtime scenario because it only has meaningful impact on init. So you'd need to launch two runtimes simultaneously and have the second one trigger off of the first one finishing and even then that'd just put us back in the race condition that we're already shipping right now.

That said it's fair to worry for any global notifications. I'm open to other suggestions on how to communicate between the runtime initialization and here. Passing a delegate all the way down through multiple thread hops seemed way too invasive.